File: backend/src/main/java/com/aplus/aplusmarket/dto/product/FindProduct.java
package com.aplus.aplusmarket.dto.product;

import com.aplus.aplusmarket.document.Products;

import com.aplus.aplusmarket.entity.Brand;
import com.aplus.aplusmarket.entity.Category;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class FindProduct {
    @Id
    private String id;
    private Brand brand;
    private Category category;
    private String name;
    private String productCode;
    private String productDetailCode;
    private double originalPrice;
    private double finalPrice;
    private String productUrl;
    private String brandName;
    private String categoryName;
    private String goodsId;

    public static FindProduct toDTO(Products products){
        return FindProduct.builder()
                .id(products.getId())
                .brandName(products.getBrand() == null  ? null : products.getBrand().getName())
                .categoryName(products.getCategory() == null ? null : products.getCategory().getCategoryName())
                .finalPrice(products.getFinalPrice())
                .originalPrice(products.getOriginalPrice())
                .productUrl(products.getProductUrl())
                .name(products.getName())
                .productCode(products.getProductCode())
                .productDetailCode(products.getProductDetailCode())
                .goodsId(products.getGoodsId())
                .build();
    }

}

File: frontend/lib/data/model/product/find_product.dart
class FindProduct {
  String? id;
  String? name;
  String? productCode;
  String? productDetailCode;
  double? originalPrice;
  double? finalPrice;
  String? productUrl;
  String? brandName;
  String? categoryName;
  String? goodsId;

  FindProduct({
    required this.id,
    required this.name,
    required this.productCode,
    required this.productDetailCode,
    required this.originalPrice,
    required this.finalPrice,
    required this.productUrl,
    required this.brandName,
    required this.goodsId,
  });

  factory FindProduct.fromMap(Map<String, dynamic> map) {
    return FindProduct(
      id: map['id'],
      name: map['name'],
      productCode: map['productCode'] as String,
      productDetailCode: map['productDetailCode'] as String,
      originalPrice: map['originalPrice'] ?? 0,
      finalPrice: map['finalPrice'] ?? 0,
      productUrl: map['productUrl'] as String? ?? null,
      brandName: map['brandName'] as String,
      goodsId: map['goodsId'] ?? null,
    );
  }

  @override
  String toString() {
    return 'FindProduct{id: $id, name: $name, productCode: $productCode, productDetailCode: $productDetailCode, originalPrice: $originalPrice, finalPrice: $finalPrice, productUrl: $productUrl, brandName: $brandName, categoryName: $categoryName}';
  }
}
